overlord/assertstate/assertstate.go

```diff
@@ -911,6 +903,56 @@ func ApplyEnforcedValidationSets(st *state.State, valsets map[string]*asserts.Va
 		valsetsTracking = append(valsetsTracking, tr)
 		valsetsSlice = append(valsetsSlice, vs)
 	}
+	return valsetsSlice, valsetsTracking, nil
+}
+
+// ApplyLocalEnforcedValidationSets enforces the supplied validation sets. It takes a map
+// of validation set keys to validation sets, pinned sequence numbers (if any),
+// installed snaps and ignored snaps. This is a 'local' variant of ApplyEnforcedValidationSets which
+// does not fetch any pre-requisites of the validation-sets. Its purpose is to be used during seeding
+// to track any included validation-sets.
+func ApplyLocalEnforcedValidationSets(st *state.State, valsets map[string]*asserts.ValidationSet, pinnedSeqs map[string]int, snaps []*snapasserts.InstalledSnap, ignoreValidation map[string]bool) error {
```

Comment:
then this should take validation sets primary keys possibly instead of valsets

Reply:
Done!